src/transport-http-server.ts

```diff
@@ -52,46 +49,77 @@ export class TransportHttpServer<BagType extends FnsBag> implements ITransport<B
             }
         }, KEEP_STALE_CONNECTIONS_FOR + 1000);
         this.onClose(() => clearInterval(staleCheckTimer));
+    }
 
-        // outgoing
-        this._app.get(this._path + 'for/:otherDeviceId', (req, res) => {
-            const otherDeviceId = req.params.otherDeviceId;
+    handler = async (req: Request): Promise<Response> => {
+        const outgoingUrlPattern = new URLPattern({ pathname: `${this._path}for/:otherDeviceId` });
+        const incomingUrlPattern = new URLPattern({ pathname: `${this._path}from/:otherDeviceId` });
+
+        if (outgoingUrlPattern.test(req.url) && req.method === 'GET') {
+            const otherDeviceId =
+                outgoingUrlPattern.exec(req.url)?.pathname.groups['otherDeviceId'];
             log('GET: (outgoing) for device', otherDeviceId);
-            if (this.isClosed) res.sendStatus(404);
 
-            const conn = this._addOrGetConnection(otherDeviceId);
+            if (!otherDeviceId || this.isClosed) {
+                return new Response('Not Found', {
+                    status: 404,
+                });
+            }
 
             const envsToSend = this._outgoingBuffer.get(otherDeviceId) ?? [];
             log(`GET: sending ${envsToSend.length} envelopes`);
             this._outgoingBuffer.set(otherDeviceId, []);
```

Comment:
This is a danger zone where the server can think it sent some envelopes and clear the buffer, but maybe the HTTP request gets interrupted and the envelopes are lost.  But I don't think the server has any way of knowing if the HTTP request was successfully completely sent back to the client...

Reply:
I don't think it's catastrophic, the other end will eventually time out and throw a timeout error if any envelopes are lost.  The sync algorithm will need to handle that anyway.

Reply:
This is mitigated a bit by the HttpClient never actually cancelling a request itself.